shared/signup/phone-number/phone-input.tsx

```diff
@@ -25,13 +33,100 @@ const menuItems = memoize((countryData, onClick) =>
 )
 
 const MenuItem = props => (
-  <Kb.Box2 direction="horizontal" centerChildren={true} style={styles.menuItem}>
-    <Kb.Text type="BodySemibold" center={true}>
-      {props.text}
-    </Kb.Text>
+  <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.menuItem}>
+    <Kb.Text type="BodySemibold">{props.text}</Kb.Text>
   </Kb.Box2>
 )
 
+type CountrySelectorProps = {
+  attachTo: () => React.Component<any> | null
+  onSelect: (arg0: string) => void
+  onHidden: () => void
+  selected: string
+  visible: boolean
+}
+
+type CountrySelectorState = {
+  selected: string
+  filter: string
+}
+
+class CountrySelector extends React.Component<CountrySelectorProps, CountrySelectorState> {
+  state = {
+    filter: '',
+    selected: this.props.selected,
+  }
+
+  componentDidUpdate(prevProps: CountrySelectorProps) {
+    if (this.props.selected !== prevProps.selected) {
+      this._onSelect(this.props.selected)
+    }
+  }
+
+  _onSelect = selected => this.setState(s => (s.selected === selected ? null : {...s, selected}))
```

Comment:
Don't spread `...s` in here, react only replaces supplied keys

Reply:
TIL!